package-res/resources/web/prompting/api/OperationAPI.js

```diff
@@ -26,9 +26,9 @@
  * @property {Object} _msgs Contains possible constants of messages
  */
 define(['common-ui/prompting/PromptPanel', 'common-ui/prompting/parameters/ParameterXmlParser'], function(PromptPanel, ParameterXmlParser) {
-  return function(api) {
+  return function(api, id) {
     this._parameterParser = new ParameterXmlParser();
-    this._promptPanel = null;
+    this._promptPanel = new PromptPanel(id);
```

Comment:
Yes, it will fail, in the case of the API. Other places can still create the prompt panel with a parameterDefinition as part of the constructor.